django/db/models/base.py

```diff
@@ -369,6 +370,15 @@ def _default_manager(cls):
         return cls._meta.default_manager
 
 
+class ModelBase(AltersDataMixin, ModelMetaclass):
+    """
+    We can't have ModelMetaclass inherit from AltersDataMixin.
+    ModelMetaclass does not pass all the attrs to it's super class.
+    So, AltersDataMixin will not be able set alters_data attribute
+    on the updated definition of subclass methods
+    """
```

Comment:
I think it would be safer to teach `add_to_class` how to deal with `data_altering_methods` itself instead as introducing an intermediate class in the type hierarchy could break third-party apps.